addons/website_event/models/event_question.py

```diff
@@ -23,12 +28,41 @@ class EventQuestion(models.Model):
                                     help="If True, this question will be asked only once and its value will be propagated to every attendees."
                                          "If not it will be asked for every attendee of a reservation.")
     is_mandatory_answer = fields.Boolean('Mandatory Answer')
+    is_identification_question = fields.Boolean('Identification Question', compute='_compute_is_identification_question')
 
     @api.constrains('event_type_id', 'event_id')
     def _constrains_event(self):
         if any(question.event_type_id and question.event_id for question in self):
             raise UserError(_('Question cannot belong to both the event category and itself.'))
 
+    @api.depends('question_type')
+    def _compute_title(self):
+        """ Since title is mandatory and to prevent the user from entering it
+        for identification questions, it will be computed automatically. But we
+        also want it to reset automatically if the question_type changes."""
+        for question in self:
+            question_type_dict = dict(self._fields['question_type']._description_selection(self.env))
+            if question.question_type not in ['simple_choice', 'text_box']:
+                question.title = question_type_dict.get(question.question_type)
+            elif question.title in question_type_dict.values():
+                question.title = False
+
+    @api.depends('question_type')
+    def _compute_is_identification_question(self):
+        for question in self:
+            question.is_identification_question = question.question_type in ['name', 'email', 'phone', 'company_name']
+
+    def init(self):
+        # Allows only 1 same identification question per event / event type
+        self.env.cr.execute("""
```

Comment:
Shoudln't we also have a constraint about "either event_id, either event_type_id" ?

Reply:
Don't you mean like this https://github.com/odoo/odoo/blob/bf772181933ce5334da35c8368455963b2478399/addons/website_event_questions/models/event_question.py#L27 ? ^^